src/TodoItem.tsx

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from './type';
+
+type Props = {
+  item: Todo,
+  onRemove: (todoId: number) => void,
+  onChangeStatus: (todo: Todo) => void,
+  onChangeTitle: (todo: Todo, newTitle: string) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  item,
+  onRemove,
+  onChangeStatus,
+  onChangeTitle,
+}) => {
+  const [editingStatus, setEditingStatus] = useState(false);
+  const [newTitle, setNewTitle] = useState(item.title);
+
+  const editTodo = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      setEditingStatus(false);
+      setNewTitle(event.target.value);
+      onChangeTitle(item, newTitle);
+
+      if (!event.target.value) {
+        onRemove(item.id);
+      }
+    }
+
+    if (event.key === 'Escape') {
+      setEditingStatus(false);
+      setNewTitle(item.title);
+    }
+  };
+
+  return (
+    <li className={classNames(
+      {
+        completed: item.completed === true,
+        editing: editingStatus,
```

Comment:
```suggestion
        completed: item.completed,
        editing: editingStatus,
```

Можно добавить здесь проверку, чтобы нельзя было редактировать тудушки, которые уже completed.

Reply:
можно, но оставлю как в демке задачи - и поменять название сделанной задачи иногда нужно)